ArduCopter/events.cpp

```diff
@@ -286,6 +286,36 @@ void Copter::failsafe_terrain_on_event()
     }
 }
 
+// check for tether fault
+void Copter::failsafe_tether_status_check()
+{
+    bool tether_trigger = copter.planck_interface.get_spool_status() == PLANCK_DECK_SPOOL_RETRIEVE;
+
+    if (tether_trigger != failsafe.tether) {
+        if (tether_trigger) {
+            failsafe_tether_on_event();
+        } else {
+            AP::logger().Write_Error(LogErrorSubsystem::FAILSAFE_TETHER, LogErrorCode::ERROR_RESOLVED);
+            failsafe.terrain = false;
+        }
+    }
+}
+
+// tether failsafe action
+// TODO: Make this handle/react differently to diffrent tether faults
+void Copter::failsafe_tether_on_event()
+{
+    failsafe.tether = true;
+    gcs().send_text(MAV_SEVERITY_CRITICAL,"Failsafe: Tether Fault"); //TODO: Make this account for diffrent tether faults
+    AP::logger().Write_Error(LogErrorSubsystem::FAILSAFE_TETHER, LogErrorCode::FAILSAFE_OCCURRED);
+
+    if((copter.flightmode != &copter.mode_planckland) && (copter.flightmode != &copter.mode_planckrtb))
```

Comment:
Is it true we _always_ want to failsafe? What about on a takeoff event?

Reply:
Yes, I think it should always trigger a failsafe. I can't think of any mode where it would be safe to do anything but land when if the BV is in RETRIEVE_ONLY. This is _especially_  true on takeoff, when it's trying to climb rapidly against a locked tether.